### PR TITLE
initial commit for editing sections

### PR DESCRIPTION
**Pull request by ilya-zhuravel**

### CODE REVIEW

File: src/app/models/config/nodes.ts [hunk 1]
```diff
@@ -1,3 +1,15 @@
+function getTitleLevel(title: Element): number {
+  let parent = title.parentNode;
+  let level = 1;
+  while (parent.nodeName !== 'body') {
```
Comment (NuclearRedeye) [inline src/app/models/config/nodes.ts:4]: Might be wise to consider also adding a check to bail out of the loop if parent or parent.nodeName are underfined/null/etc.

**Review by NuclearRedeye — COMMENTED** [review]
I'm assuming from the title that you still have more work to do on this PR, as such simply just adding my comments for now.
I tested it locally and it's working well, noticed a few minor issues that you might want to look at whilst you're in the area or perhaps we can create some new issues to capture them.
1. Copy and Paste looses the formatting, so trying to move a title around resets it back to plain text. This might be a limitation that we can live with for now to be honest.
2. The 'Paragraph' option in the 'Paragraph' menu seems to always be enabled when controls that don't support it are focused, e.g. the abstract or impact statement. Trying to use it appears to not do anything, so it's not causing any major issues.

**Comment by ilya-zhuravel** [comment]
> I'm assuming from the title that you still have more work to do on this PR, as such simply just adding my comments for now.
> I tested it locally and it's working well, noticed a few minor issues that you might want to look at whilst you're in the area or perhaps we can create some new issues to capture them.
> 
> 1. Copy and Paste looses the formatting, so trying to move a title around resets it back to plain text. This might be a limitation that we can live with for now to be honest.
So copy and paste we gonna address in a different ticket. Without this it won't work.
> 2. The 'Paragraph' option in the 'Paragraph' menu seems to always be enabled when controls that don't support it are focused, e.g. the abstract or impact statement. Trying to use it appears to not do anything, so it's not causing any major issues.
That's a bug

**Review by NuclearRedeye — APPROVED** [review]

**Comment by NuclearRedeye** [comment]
No problem, i'll approve.

**Pull request merged**